main.star

```diff
@@ -1,40 +1,50 @@
-MONGO_DB_PROTOCOL_NAME = "mongodb"
-MONGO_DB_PORT_ID = "mongodb"
+NAME_ARG = "name"
+USER_ARG = "user"
+PASSWORD_ARG = "password"
+IMAGE_ARG = "image"
+ENV_VARS_ARG = "env_vars"
 
-MONGO_DB_DEFAULT_PORT_NUMBER = 27017
-MONGO_DB_PORT_NUMBER_ARG = "MONGO_DB_PORT_NUMBER"
-
-MONGO_DB_DEFAULT_IMAGE_TAG = "mongo:6.0.5"
-MONGO_DB_IMAGE_TAG_ARG = "MONGO_DB_IMAGE_TAG"
-
-ENV_VARS_ARG_NAME = "env_vars"
+PORT_NAME = "mongodb"
+PORT_NUMBER = 27017
+PROTOCOL_NAME = "mongodb"
 
 def run(plan, args):
-    mongo_db_image = MONGO_DB_DEFAULT_IMAGE_TAG
-    if MONGO_DB_IMAGE_TAG_ARG in args:
-        mongo_db_image = args[MONGO_DB_IMAGE_TAG_ARG]
-
-    mongo_db_port_number = MONGO_DB_DEFAULT_PORT_NUMBER
-    if MONGO_DB_PORT_NUMBER_ARG in args:
-        mongo_db_port_number = args[MONGO_DB_PORT_NUMBER_ARG]
-
-    env_vars = {}
-    if ENV_VARS_ARG_NAME in args:
-        env_vars = args[ENV_VARS_ARG_NAME]
-
-    #Add a mongo db server
-    mongo_db_service = plan.add_service(
-        name="mongoDB",
+    service_name = args.get(NAME_ARG, "mongoDB")
+    image = args.get(IMAGE_ARG, "mongo:6.0.5")
+    user = args.get(USER_ARG, "root")
+    password = args.get(PASSWORD_ARG, "password")
+    env_var_overrides = args.get(ENV_VARS_ARG, {})
+
+    env_vars = {
+        "MONGO_INITDB_ROOT_USERNAME": user,
+        "MONGO_INITDB_ROOT_PASSWORD": password,
+    }
+    env_vars |= env_var_overrides
+
+    # Add the server
+    service = plan.add_service(
+        name=service_name,
         config=ServiceConfig(
-            image=mongo_db_image,
+            image=image,
             ports={
-                MONGO_DB_PORT_ID: PortSpec(
-                    number=mongo_db_port_number,
-                    application_protocol=MONGO_DB_PROTOCOL_NAME
+                PORT_NAME: PortSpec(
```

Comment:
Did this in my Postgres package as well - I thiiiink I'd like to soft-migrate to "port name" rather than "port ID". "Port ID" begs this question "what's the unique identification?" (and we don't have one)